src/Controller/AuthorizationController.php

```diff
@@ -56,27 +57,46 @@
      */
     public function __invoke(ServerRequestInterface $request): ResponseInterface
     {
-        $authorizationRequest = $this->authorizationServer->validateAuthorizationRequest($request);
+        $queryParameters = $request->getQueryParams();
+        $state = null;
 
-        $user = $this->authenticationService->getAuthenticateUser($request);
+        if (!isset($queryParameters[ProcessingChain::AUTHPARAM])) {
+            $authorizationRequest = $this->authorizationServer->validateAuthorizationRequest($request);
+            $state = $this->authenticationService->processRequest($request, $authorizationRequest);
+            // processState will trigger a redirect
+        }
+
+        $state ??= $this->authenticationService->loadState($queryParameters);
+        $authorizationRequest = $this->authenticationService->getAuthorizationRequestFromState($state);
+
+        $user = $this->authenticationService->getAuthenticateUser($state);
 
         $authorizationRequest->setUser($user);
         $authorizationRequest->setAuthorizationApproved(true);
 
-        if ($authorizationRequest instanceof AuthorizationRequest) {
-            $authorizationRequest->setIsCookieBasedAuthn($this->authenticationService->isCookieBasedAuthn());
-            $authorizationRequest->setAuthSourceId($this->authenticationService->getAuthSourceId());
-            $authorizationRequest->setSessionId($this->authenticationService->getSessionId());
+        $authorizationRequest->setIsCookieBasedAuthn($this->authenticationService->isCookieBasedAuthn());
```

Comment:
Note that this method does not exist in OAuth2AuthorizationRequest (League\OAuth2\Server\RequestTypes\AuthorizationRequest). Previously there was a check if the instance is of expected type. The whole thing "stinks", but that's until we introduce proper OAuth2 package briges... :/

Reply:
The check is no longer needed since we do it inside the AuthorizationController::getAuthorizationRequestFromState.

Reply:
I fixed it. Even though i was checking the instance of the authorizationRequest, i was not taking into account the 
```php
        return $this->authorizationServer->completeAuthorizationRequest(
            $authorizationRequest,
            $this->psrHttpBridge->getResponseFactory()->createResponse(),
        );
```
code